pkg/operator/api/userconfig/apis.go

```diff
@@ -357,162 +296,128 @@ func (apis APIs) Validate(deploymentName string, projectFileMap map[string][]byt
 	return nil
 }
 
-func (tf *TensorFlow) Validate(projectFileMap map[string][]byte) error {
-	awsClient, err := aws.NewFromS3Path(tf.Model, false)
-	if err != nil {
-		return err
-	}
-	if strings.HasSuffix(tf.Model, ".zip") {
-		if ok, err := awsClient.IsS3PathFile(tf.Model); err != nil || !ok {
-			return errors.Wrap(ErrorExternalNotFound(tf.Model), TensorFlowKey, ModelKey)
-		}
-	} else {
-		path, err := GetTFServingExportFromS3Path(tf.Model, awsClient)
-		if path == "" || err != nil {
-			return errors.Wrap(ErrorInvalidTensorFlowDir(tf.Model), TensorFlowKey, ModelKey)
-		}
-		tf.Model = path
-	}
-	if _, ok := projectFileMap[tf.Predictor]; !ok {
-		return errors.Wrap(ErrorImplDoesNotExist(tf.Predictor), TensorFlowKey, PredictorKey)
-	}
-	if tf.PythonPath != nil {
-		if err := ValidatePythonPath(*tf.PythonPath, projectFileMap); err != nil {
-			return errors.Wrap(err, TensorFlowKey)
-		}
-	}
-	return nil
-}
-
-func (tf *TensorFlow) UserConfigStr() string {
+func (predictor *Predictor) UserConfigStr() string {
 	var sb strings.Builder
-	sb.WriteString(fmt.Sprintf("%s: %s\n", ModelKey, tf.Model))
-	sb.WriteString(fmt.Sprintf("%s: %s\n", PredictorKey, tf.Predictor))
-	if tf.PythonPath != nil {
-		sb.WriteString(fmt.Sprintf("%s: %s\n", PythonPathKey, *tf.PythonPath))
+	sb.WriteString(fmt.Sprintf("%s: %s\n", TypeKey, predictor.Type))
+	sb.WriteString(fmt.Sprintf("%s: %s\n", PathKey, predictor.Path))
+	if predictor.Model != nil {
+		sb.WriteString(fmt.Sprintf("%s: %s\n", ModelKey, *predictor.Model))
 	}
-	if tf.SignatureKey != nil {
-		sb.WriteString(fmt.Sprintf("%s: %s\n", SignatureKeyKey, *tf.SignatureKey))
+	if predictor.PythonPath != nil {
+		sb.WriteString(fmt.Sprintf("%s: %s\n", PythonPathKey, *predictor.PythonPath))
 	}
-	if len(tf.Config) > 0 {
+	if len(predictor.Config) > 0 {
 		sb.WriteString(fmt.Sprintf("%s:\n", ConfigKey))
-		d, _ := yaml.Marshal(&tf.Config)
+		d, _ := yaml.Marshal(&predictor.Config)
 		sb.WriteString(s.Indent(string(d), "  "))
 	}
-	if len(tf.Env) > 0 {
+	if len(predictor.Env) > 0 {
 		sb.WriteString(fmt.Sprintf("%s:\n", EnvKey))
-		d, _ := yaml.Marshal(&tf.Env)
+		d, _ := yaml.Marshal(&predictor.Env)
 		sb.WriteString(s.Indent(string(d), "  "))
 	}
+	if predictor.SignatureKey != nil {
```

Comment:
nit: I think this could go above env and config, right below model, but not a big deal